pkg/apis/aws/validation/worker.go

```diff
@@ -86,6 +86,44 @@ func ValidateWorkerConfig(workerConfig *apisaws.WorkerConfig, volume *core.Volum
 	return allErrs
 }
 
+func ValidateWorkersAgainstCloudProfileOnCreation(workers []core.Worker, region string, awsCloudProfile *apisaws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, validateWorkersConfigAgainstCloudProfile(workers, region, awsCloudProfile, fldPath)...)
+
+	return allErrs
+}
+
+func ValidateWorkersAgainstCloudProfileOnUpdate(_, newWorkers []core.Worker, region string, awsCloudProfile *apisaws.CloudProfileConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Validate the existence of the images the new/updated workers are to use. Validating the images used by old workers is not possible at this point, as they might
+	// have been removed from the CloudProfile already.
+	allErrs = append(allErrs, validateWorkersConfigAgainstCloudProfile(newWorkers, region, awsCloudProfile, fldPath)...)
```

Comment:
I am not sure this is okay. Indeed the machine images may be gone from a cloudprofile, but we do not want to prevent other updates to the shoot spec because of that. 

Hence the set of workers that can be validated with this image checking is only the subnet of workers whose image we tried to update (or maybe even a general update to them) or new workers.

So to recap, the newWorkers here should contain the subnet of workers that:
a) are new (`not existing in oldWorkers`)
b) images changes (`newWorkers exists in oldWorkers` && `newWorker.Image != oldWorker.Image`)

WDYT?